source/slang-glslang/slang-glslang.cpp

```diff
@@ -86,6 +89,119 @@ static void dumpDiagnostics(
     dump(log.c_str(), log.length(), request->diagnosticFunc, request->diagnosticUserData, stderr);
 }
 
+// Apply the SPIRV-Tools optimizer to generated SPIR-V based on the desired optimization level
+// TODO: add flag for optimizing SPIR-V size as well
+static void glslang_optimizeSPIRV(std::vector<unsigned int>& spirv, unsigned optimizationLevel, unsigned debugInfoType)
+{
+    spv_target_env target_env = SPV_ENV_UNIVERSAL_1_2;
+
+    spvtools::Optimizer optimizer(target_env);
+    optimizer.SetMessageConsumer(
+        [](spv_message_level_t level, const char *source, const spv_position_t &position, const char *message) {
+        auto &out = std::cerr;
+        switch (level)
+        {
+        case SPV_MSG_FATAL:
+        case SPV_MSG_INTERNAL_ERROR:
+        case SPV_MSG_ERROR:
+            out << "error: ";
+            break;
+        case SPV_MSG_WARNING:
+            out << "warning: ";
+            break;
+        case SPV_MSG_INFO:
+        case SPV_MSG_DEBUG:
+            out << "info: ";
+            break;
+        default:
+            break;
+        }
+        if (source)
+        {
+            out << source << ":";
+        }
+        out << position.line << ":" << position.column << ":" << position.index << ":";
+        if (message)
+        {
+            out << " " << message;
+        }
+        out << std::endl;
```

Comment:
This isn't how most of our error messages are formatted, but it probably isn't the end of the world to leave it like this for now. I think we tend to do `<file>(<line>): <severity>: <message>`. I don't recall if we put in the column number by default, but I probably wouldn't put both column and index.

Reply:
No problem, I'll make a new MR to clean this up. I brought the format over from spirv-opt.